client/src/utils/userContext.js

```diff
@@ -1,20 +1,29 @@
 import { useContext, createContext, useState, useEffect } from "react";
 import { UseIsLoggedIn } from "./authenticate";
+import jwt_decode from "jwt-decode";
 
 const UserContext = createContext();
 
 export const useUserContext = () => useContext(UserContext);
 
 export const UserProvider = ({ children }) => {
   const [userLogged, setUserLogged] = useState(UseIsLoggedIn());
+  const userToken = localStorage.getItem("auth_token");
```

Comment:
Look at changing the local storage logic to be in a useEffect with an empty dependency array